src/sipmpdf/functions.py

```diff
@@ -216,47 +245,384 @@ def darkcurrent_response(x: np.float64,
+  """
+
+  kern = kernel_switch(x, pedestal, gain, common_noise, pixel_noise,
+                       poisson_mean, poisson_borel, ap_prob, ap_beta, dc_prob,
+                       dc_res)
+  dc_smear = kern.sqrt(common_noise**2 + pixel_noise**2)
+
+  no_pe_discharge_response = generalized_poisson(
+    k=kern.zeros_like(x), mean=poisson_mean,
+    borel=poisson_borel) * (1 - dc_prob) * normal(
+      x=x, mean=pedestal,
+      scale=common_noise) + dc_prob * darkcurrent_response_smeared(
+        x=x, smear=dc_smear, gain=gain, resolution=dc_res)
+  pe_discharge_response = k_summation(x=x,
+                                      common_noise=common_noise,
+                                      pixel_noise=pixel_noise,
+                                      ap_beta=ap_beta,
+                                      ap_prob=ap_prob,
+                                      pedestal=pedestal,
+                                      gain=gain,
+                                      poisson_mean=poisson_mean,
+                                      poisson_borel=poisson_borel)
+
+  return no_pe_discharge_response + pe_discharge_response
+
+
+def sipm_response_no_dark(x: np.float64,
+                          pedestal: np.float64,
+                          gain: np.float64,
+                          common_noise: np.float64,
+                          pixel_noise: np.float64,
+                          poisson_mean: np.float64,
+                          poisson_borel: np.float64,
+                          ap_prob: np.float64,
+                          ap_beta: np.float64,
+                          dc_res: np.float64 = 1e-4) -> np.float64:
+  """
+  Main function for the general SiPM low light response probability function not accounting for dark current.
+  Using numba for calculation acceleration (the function is difficult to express
+  in pure array syntax)
+  
+  Parameters
+  ----------
+  x : numpy.float64
+      The observable value
+  pedestal : np.float64
+      The pedestal value
+  gain : np.float64
+      The gain of the SiPM
+  common_noise : np.float64
+      Common factor noise corresponding to sigma_0
+  pixel_noise : np.float64
+      Average pixel gain variation corresponding to sigma_1
+  poisson_mean : np.float64
+      Mean optical power of PE discharges
+  poisson_borel : np.float64
+      Poisson borel value
+  ap_beta : np.float64
+      Afterpulse timescale factor.
+  ap_prob : np.float64
+      Probability of afterpulsing
+  dc_prob : np.float64
+      Probability of dark current
+  dc_res : np.float64, optional
+      resolution factor, relative to the SiPM gain factor, by default 1e-4
+  Returns
+  -------
+  numpy.float64
+      SiPM response without dark current probability density function as a function of x
   """
-  pass
+  kern = kernel_switch(x, pedestal, gain, common_noise, pixel_noise,
+                       poisson_mean, poisson_borel, ap_prob, ap_beta, dc_res)
+  dc_prob = kern.zeros_like(x)
```

Comment:
Try and implement this using `_k_summation` and `_afterpulse_summation` only, as the dark current response has the most expensive dimension expansion (factor of 1000!) So users might want to fall back to a method witout dark current reponse if something goes wrong.